addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -41,9 +41,12 @@ class MyModel(models.Model):
                 'Field "(field)r on model %(model)r must be of type Many2one and have tracking=True for the computation of duration.',
                 field=self._track_duration_field, model=self._name
             ))
-
-        self.env['mail.tracking.value'].flush_model()
-        self.env['mail.message'].flush_model()
+        if f'mail.tracking.{self._name}' in self.env.cr.precommit.data:
+            # this will generate the tracking messages and flush all
+            self._track_finalize()
+        else:
+            self.env['mail.tracking.value'].flush_model()
+            self.env['mail.message'].flush_model()
```

Comment:
maybe it's more clear like currently, but couldn't this code be equivalent:

```suggestion
        self._track_finalize()
        self.env['mail.tracking.value'].flush_model()
        self.env['mail.message'].flush_model()
```
since `_track_finalize` does nothing if there is no `mail.tracking.{self._name}` in precommit data, and does flush_all if there is but then doing flush_model should do nothing.